### PR TITLE
Autotype

### PR DESCRIPTION
**Pull request by natoverse**
Reworks the table loading to support our internal autotyping. This also resolves the issues with papaparse, so all csv parse options are now enabled in the UI. The primary table reading function, `readTable`, now requires a DataTableSchema so it can fully interpret the content.

### CODE REVIEW

File: javascript/app-framework/src/components/tables/ImportTable/ImportTable.tsx [hunk 1]
```diff
@@ -45,6 +53,14 @@ export const ImportTable: React.FC<ImportTableProps> = memo(
 
 		const [name, setName] = useState<string>(removeExtension(file.name ?? ''))
 
+		const [rawContent, setRawContent] = useState<string | undefined>()
+		useEffect(() => {
+			const f = async () => {
```
Comment (darthtrevino) [inline javascript/app-framework/src/components/tables/ImportTable/ImportTable.tsx:58]: This short callback may be clearer as a promise chain

`file.toText().then(setRawContent)`

File: javascript/tables/src/__tests__/data/companies-headerless.csv [hunk 2]
```diff
@@ -1,5 +1,6 @@
 1,Microsoft,160000,true
 2,Apple,150000,true
 3,Google,135000,true
+
```
Comment (darthtrevino) [inline javascript/tables/src/__tests__/data/companies-headerless.csv:4]: Is a blank line here intentional?

File: javascript/tables/src/readTable.ts [hunk 3]
```diff
@@ -17,23 +23,105 @@ import { readJSONTable } from './readJSONTable.js'
  * @returns
  */
 export async function readTable(
-	input: Blob | undefined,
-	schema: DataTableSchema,
+	input: Blob | string | undefined,
+	schema: Partial<DataTableSchema>,
+	options?: {
```
Comment (darthtrevino) [inline javascript/tables/src/readTable.ts:28]: This should be extracted into an interface

**Review by darthtrevino — APPROVED** [review]

File: javascript/tables/src/__tests__/data/companies-headerless.csv (hunk 2 above)
Comment (natoverse) [reply javascript/tables/src/__tests__/data/companies-headerless.csv:4]: yep, it tests that the parser ignores blank lines

**Review by monik182 — APPROVED** [review]

**Pull request merged**